ocs_ci/ocs/cephfs_workload.py

```diff
@@ -0,0 +1,216 @@
+            # cluster configuration or unrelated bug which must have happened
+            # before this test case
+            error_msg = "unexpected problem with start of the workload, cluster is either misconfigured or broken"
+            logger.exception(error_msg)
+            logger.debug(self.workload_file.describe())
+            raise exceptions.UnexpectedBehaviour(error_msg) from ex
+
+    def fetch_and_validate_data(self):
+        """
+        While the workload is running, try to validate the data
+        from the cephfs volume of the workload.
+
+        Raise:
+            NotFoundError: When the given volume is not found in given spec
+            Exception: When the data verification job failed
+
+        """
+        # if no obvious problem was detected, run the logreader job to validate
+        # checksums in the log files (so that we are 100% sure that nothing went
+        # wrong with the IO or the data)
+        with open(constants.LOGWRITER_CEPHFS_READER, "r") as job_file:
+            job_dict = yaml.safe_load(job_file.read())
+        # if we are running in disconnected environment, we need to mirror the
+        # container image first, and then use the mirror instead of the original
+        if config.DEPLOYMENT.get("disconnected"):
+            update_container_with_mirrored_image(self.deploy_dict["spec"]["template"])
+        # drop topology spread constraints related to zones
+        topology.drop_topology_constraint(
+            job_dict["spec"]["template"]["spec"], topology.ZONE_LABEL
+        )
+        # we need to match number of jobs with the number used in the workload
+        job_dict["spec"]["completions"] = self.deploy_dict["spec"]["replicas"]
+        job_dict["spec"]["parallelism"] = self.deploy_dict["spec"]["replicas"]
+        # and reffer to the correct pvc name
+        try:
+            link_spec_volume(
+                job_dict["spec"]["template"]["spec"],
+                "logwriter-cephfs-volume",
+                self.pvc_dict["metadata"]["name"],
+            )
+        except Exception as ex:
+            error_msg = "LOGWRITER_CEPHFS_READER no longer matches code of this test"
+            raise Exception(error_msg) from ex
+        # prepare k8s yaml file for the job
+        job_file = ObjectConfFile("log_reader", [job_dict], self.project, self.tmp_path)
+        # deploy the job, starting the log reader pods
+        logger.info(
+            "starting log reader data validation job to fully check the log data",
+        )
+        job_file.create()
+        # wait for the logreader job to complete (this should be rather quick)
+        try:
+            job.wait_for_job_completion(
+                job_name=job_dict["metadata"]["name"],
+                namespace=self.project.namespace,
+                timeout=300,
+                sleep_time=30,
+            )
+        except exceptions.TimeoutExpiredError:
```

Comment:
Why are we raising a `TimeoutExpiredError` with a log message like `"verification failed to complete in time: data loss or broken cluster?"`?
How to debug in case of an actual data loss or a broken cluster? The analysis will always result in `TimeoutExpiredError`.

Reply:
Do you suggest raising another exception? Maybe just throwing `raise Exception(error_msg)` instead of `logger.exception(error_msg)`

Reply:
Whatever you feel is the best, my concern is how do we debug and conclude if more time is needed in such scenarios where a failure has occurred with timeoutexception? What if the reason is something else where increasing the timeout wouldn't help, for e.g. a broken cluster? Is it expected to recover by increasing the timeout?